imagorpath/params.go

```diff
@@ -20,33 +20,37 @@ type Filters []Filter
 
 // Params image endpoint parameters
 type Params struct {
-	Params        bool    `json:"-"`
-	Path          string  `json:"path,omitempty"`
-	Image         string  `json:"image,omitempty"`
-	Unsafe        bool    `json:"unsafe,omitempty"`
-	Hash          string  `json:"hash,omitempty"`
-	Meta          bool    `json:"meta,omitempty"`
-	Trim          bool    `json:"trim,omitempty"`
-	TrimBy        string  `json:"trim_by,omitempty"`
-	TrimTolerance int     `json:"trim_tolerance,omitempty"`
-	CropLeft      int     `json:"crop_left,omitempty"`
-	CropTop       int     `json:"crop_top,omitempty"`
-	CropRight     int     `json:"crop_right,omitempty"`
-	CropBottom    int     `json:"crop_bottom,omitempty"`
-	FitIn         bool    `json:"fit_in,omitempty"`
-	Stretch       bool    `json:"stretch,omitempty"`
-	Width         int     `json:"width,omitempty"`
-	Height        int     `json:"height,omitempty"`
-	PaddingLeft   int     `json:"padding_left,omitempty"`
-	PaddingTop    int     `json:"padding_top,omitempty"`
-	PaddingRight  int     `json:"padding_right,omitempty"`
-	PaddingBottom int     `json:"padding_bottom,omitempty"`
-	HFlip         bool    `json:"h_flip,omitempty"`
-	VFlip         bool    `json:"v_flip,omitempty"`
-	HAlign        string  `json:"h_align,omitempty"`
-	VAlign        string  `json:"v_align,omitempty"`
-	Smart         bool    `json:"smart,omitempty"`
-	Filters       Filters `json:"filters,omitempty"`
+	Params            bool    `json:"-"`
+	Path              string  `json:"path,omitempty"`
+	Image             string  `json:"image,omitempty"`
+	Unsafe            bool    `json:"unsafe,omitempty"`
+	Hash              string  `json:"hash,omitempty"`
+	Meta              bool    `json:"meta,omitempty"`
+	Trim              bool    `json:"trim,omitempty"`
+	TrimBy            string  `json:"trim_by,omitempty"`
+	TrimTolerance     int     `json:"trim_tolerance,omitempty"`
+	CropLeft          int     `json:"crop_left,omitempty"`
+	CropLeftPercent   float64 `json:"crop_left_percent,omitempty"`
```

Comment:
What do you think about e.g. CropLeft and CropLeftPercent to combine as 1 value

- value < 1: evaluate as percentage
- value >= 1: evaluate as pixel

seems viable?

Reply:
Though we will then need to deal with floating points more carefully, when evaluating pixels.

Reply:
Seems viable to me. I can try it and see if floating points is not messing up anything.